### ISSUE TITLE
OBS Studio Music Edition Master Mixer monitoring

### ISSUE THREAD

**Issue by jesuslopezuy** [pkviet/obs-studio#38]
Hello pkviet!
I report: OBS Studio Music Edition Master Mixer monitoring, not working.
Best regards from Uruguay!

**Comment by x0rx** [comment]
Same here on 26.0.0 release. Click on monitor icon does nothing.

**Comment by FelipeJz** [comment]
I have the same problem, monitoring is just not working :( @pkviet

**Comment by pkviet** [comment]
i'll definitely need more information because it's working on my side.
So the minimum would be to post logs.

**Comment by jesuslopezuy** [comment]
Hi pkviet!
Attached log file.
Cordially, Jesus.
[2020-10-16 09-32-44.txt](https://github.com/pkviet/obs-studio/files/5391916/2020-10-16.09-32-44.txt)

**Comment by x0rx** [comment]
Sorry, here is mine. There's a problem coming from 25.0.8 also. If monitor input mixer icons are activated, switching to master mixer monitoring disables icons on input mixer but not the 
[2020-10-16 16-18-20.txt](https://github.com/pkviet/obs-studio/files/5392287/2020-10-16.16-18-20.txt)
sound.

**Comment by pkviet** [comment]
@x0rx  what's that 96 kHz device ? which device are you using to monitor  (the one set in Settings > audio > monitoring device) ?
@jesuslopezuy same questions.
I notice that both your windows versions are older than the one I targeted during compilation (1903).
Try installing the last redistributable from microsoft: https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads
Tell me if it makes any difference. And post the new logs. It's not related to monitoring but one of the encoder dll fails to load.

**Comment by x0rx** [comment]
OBS settings:

Global devices:
- USB DAC, the ODAC used for speakers (Desktop1).
- a Realtek Digital Output used as fake output for Desktop2. 
- Realtek mic at front panel

Monitoring device:
- Realtek HD Audio output at front panel

I'm using Win 10 LTSC x64 (1809)
Updated Visual C++ 2015-2019. Rebooted and no master mixer monitoring.

[2020-10-16 20-52-50.txt](https://github.com/pkviet/obs-studio/files/5393795/2020-10-16.20-52-50.txt)

**Comment by pkviet** [comment]
@x0rx  ok try the build from here:
https://github.com/pkviet/obs-studio/actions/runs/289971644
Get the zip, unzip in any folder. It will use the same settings. If it doesn't work, you just put the unzipped folder in the trash.

**Comment by x0rx** [comment]
Tested. It´s version 25.0.7.777 ?
It doesn´t have the Master Mixer panel.

**Comment by pkviet** [comment]
> Tested. It´s version 25.0.7.777 ?
> It doesn´t have the Master Mixer panel.

oops sorry, wrong build.
Try this one:
https://github.com/pkviet/obs-studio/actions/runs/281018862

**Comment by x0rx** [comment]
Sorry. Master Mixer monitoring not working.

**Comment by pkviet** [comment]
@x0rx  but monitoring on the input source mixer is working ? just the master mixer is not ?
edit: ok i can repro; it works on earlier versions but indeed i broke it. I should be able to fix it soon. lemme investigate

**Comment by x0rx** [comment]
Correct. Problem is only with monitoring on/off in master mixer.
Also if you can take a look at direct monitoring switching from/to input mixer to master mixer. (This is not working in 25.0.8 and can´t test this in 26.0 until first problem is solved).
Is nice not to have to power off each one of input mixer icons before activating the master mixer one.

Any request to test anything, please tell me.

By the way, thank you very much for all this work!!

**Comment by pkviet** [comment]
ok i bisected the issue; it's to do with hotkeys that I had enabled for monitoring. 
edit: nope it's another commit

**Comment by pkviet** [comment]
Should be fixed in the build which is being compiled here:
https://github.com/pkviet/obs-studio/actions/runs/311649249
(at the time of writing, they're still being compiled so wait until you see the zip listed on the page; on my dev machine the bug was fixed though).
Test and report whether it's fixed for you or not.
edit: github has issues at the moment with CI; I'll upload a test build rather

**Comment by pkviet** [comment]
test build here: https://github.com/pkviet/obs-studio/releases/tag/test26.0.2

**Comment by x0rx** [comment]
Fixed. Let's see other people tests... thanks to @jesuslopezuy for opening the issue.
What do you think about the other issue? Better to open other thread?
Thanks @pkviet !!

**Comment by pkviet** [comment]
@x0rx  open another issue and detail what's the problem, I didn't get it clearly

**Comment by x0rx** [comment]
ok. May I ask you a question: Latency between input mixer and master mixer is direct result of processing load by filter plugins? I´m concerned about if this can have an impact in final a/v sync.

**Comment by FelipeJz** [comment]
Works great now! Thanks @pkviet

**Comment by pkviet** [comment]
@x0rx The filters have negligible impact. Don't worry about them. Obs takes it into account to sync video to audio anyway.

**Comment by jesuslopezuy** [comment]
Hi pkviet!
1- I installed the last redistributable from microsoft: https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads, and the error was not corrected.
2- I downloaded the test build https://github.com/pkviet/obs-studio/releases/download/test26.0.2/RelWithDebInfo.7z, and the error was corrected Master Mixer monitoring.
Very grateful for the prompt response and total dedication to help us!!!

Cordially,
@jesuslopezuy

Note: very grateful to everyone for supporting this Project!
